src/components/SlateEditor/plugins/toolbar/SlateToolbar.tsx

```diff
@@ -137,78 +109,86 @@ const SlateToolbar = (props: Props) => {
-    : topicArticleElements;
-
-  const markButtons = toolbarElements.mark.map((type) => (
-    <ToolbarButton
-      key={type}
-      type={type}
-      kind={'mark'}
-      isActive={isMarkActive(editor, type)}
-      handleOnClick={(event: MouseEvent, kind: string, type: string) => {
-        onButtonClick(event, editor, kind, type);
-      }}
-    />
-  ));
-
-  const blockButtons = toolbarElements.block.map((type) => (
-    <ToolbarButton
-      key={type}
-      type={type}
-      kind={'block'}
-      isActive={
-        type.includes('list')
-          ? isActiveList(editor, type)
-          : hasNodeWithProps(editor, specialRules[type] ?? { type })
+  });
+
+  const onButtonClick = useCallback(
+    (event: MouseEvent, kind: string, type: string) => {
+      if (kind === 'mark') {
+        toggleMark(event, editor, type);
+      } else if (kind === 'block') {
+        handleClickBlock(event, editor, type);
+      } else if (kind === 'inline') {
+        handleClickInline(event, editor, type);
+      } else if (kind === 'table') {
+        handleClickTable(event, editor, type);
       }
-      handleOnClick={(event: MouseEvent, kind: string, type: string) => {
-        onButtonClick(event, editor, kind, type);
-      }}
-    />
-  ));
-  const inlineButtons = toolbarElements.inline.map((type) => (
-    <ToolbarButton
-      key={type}
-      type={type}
-      kind={'inline'}
-      isActive={hasNodeWithProps(editor, specialRules[type] ?? { type })}
-      handleOnClick={(event: MouseEvent, kind: string, type: string) => {
-        onButtonClick(event, editor, kind, type);
-      }}
-    />
-  ));
-
-  const tableCellElement = getCurrentBlock(editor, TYPE_TABLE_CELL)?.[0];
-  const tableButtons =
-    !!tableCellElement &&
-    toolbarElements.table?.map((type, index) => (
-      <ToolbarButton
-        key={index}
-        type={type}
-        kind={'table'}
-        isActive={hasCellAlignOfType(editor, type)}
-        handleOnClick={(event: MouseEvent, kind: string, type: string) => {
-          onButtonClick(event, editor, kind, type);
-        }}
-      />
-    ));
+    },
+    [editor],
+  );
+
+  const isActiveList = useCallback(
+    (type: string) => {
+      if (type === 'definition-list') {
+        const path = getCurrentBlock(editor, TYPE_DEFINITION_LIST)?.[1];
+        if (path) {
+          return hasDefinitionListItem(editor);
+        }
+        return false;
+      }
+      return hasListItem(editor, type);
+    },
+    [editor],
+  );
+
+  const onMouseDown = useCallback((e: MouseEvent) => e.preventDefault(), []);
 
   return (
     <Portal>
-      <ToolbarContainer ref={portalRef}>
-        {markButtons}
-        {blockButtons}
-        {inlineButtons}
-        {tableButtons}
+      <ToolbarContainer ref={portalRef} onMouseDown={onMouseDown}>
+        {toolbarElements.mark.map((type) => (
+          <ToolbarButton
+            key={type}
+            type={type}
+            kind={'mark'}
```

Comment:
Disse kan vel defineres slik.
```suggestion
            kind="mark"
```